src/network.rs

```diff
@@ -240,21 +269,44 @@ impl NetworkBehaviourEventProcess<orderbook::BehaviourOutEvent> for Nectar {
                 response_channel,
                 order_id,
             } => {
+                let taker = Taker::new(taker_peer_id);
+                let ongoing_trade_with_taker_exists = match self.active_takers.contains(&taker) {
```

Comment:
Neat! So, we only bubble the event up to handle it in the maker if there is no active trade with this taker.

Reply:
Yes! It took me quite long to reach this design, but I think it makes sense to move this logic here. And we get the benefit that you describe :)